src/r5py/r5/transport_network.py

```diff
@@ -110,6 +122,43 @@ def __exit__(self, exception_type, exception_value, traceback):
         """Exit context."""
         return False
 
+    @functools.cached_property
+    def _cache_directory(self):
+        cache_dir = (
+            pathlib.Path(config.CACHE_DIR) / f"{self.__class__.__name__:s}_{hash(self):x}"
```

Comment:
this creates directory names looking like `CACHE_DIR/TransportNetwork_1a23b4c5/` (the hash part *should* be unique during one run, as it is derived from, among other things, the memory address of the TransportNetwork instance)